# GitHub Issues overview Dataset

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

##### Load the dataset

In [2]:
df = pd.read_csv('archive/a_github_issues_overview_dataset.csv')

#### Check Number of Rows and Columns

In [3]:
df.shape

(15955, 16)

#### Check Present Columns In Dataset

In [4]:
print('These are the columns which are present in our dataset')
i = 1
for column in df.columns:
    print(i,'.', column)
    i += 1
del i

These are the columns which are present in our dataset
1 . repo_name
2 . title
3 . body
4 . user_login
5 . created_at
6 . answer_1
7 . answer_2
8 . answer_3
9 . answer_4
10 . answer_5
11 . label_1
12 . label_2
13 . label_3
14 . label_4
15 . label_5
16 . likes


#### Check if any empty row is present

In [5]:
df.isna().sum()

repo_name         0
title             0
body              0
user_login        0
created_at        0
answer_1          0
answer_2       3447
answer_3       6949
answer_4       9520
answer_5      11442
label_1           9
label_2       15951
label_3        7517
label_4       12172
label_5       14409
likes             0
dtype: int64

**Note:** `This dataset contain alot number of empty rows, total rows are 15955 and one column has 15951 empty rows`

#### Get basic Info of Dataset

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15955 entries, 0 to 15954
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   repo_name   15955 non-null  object
 1   title       15955 non-null  object
 2   body        15955 non-null  object
 3   user_login  15955 non-null  object
 4   created_at  15955 non-null  object
 5   answer_1    15955 non-null  object
 6   answer_2    12508 non-null  object
 7   answer_3    9006 non-null   object
 8   answer_4    6435 non-null   object
 9   answer_5    4513 non-null   object
 10  label_1     15946 non-null  object
 11  label_2     4 non-null      object
 12  label_3     8438 non-null   object
 13  label_4     3783 non-null   object
 14  label_5     1546 non-null   object
 15  likes       15955 non-null  int64 
dtypes: int64(1), object(15)
memory usage: 1.9+ MB


#### As label_2 have only two not null values, Let's See them

In [7]:
# Filter rows where label_2 is not null
filtered_df = df[df['label_2'].notnull()]
print('Values which are not null in label_2 are')
i = 1
for row in filtered_df['label_2']:
    print(f'{i}. {row}')
del i

Values which are not null in label_2 are
1. Area: Design/Frontend
1. stale issue
1. Area: Design/Frontend
1. Component: Ui


#### Analyze the dataset by seeing its values

In [8]:
df.head(4)

repo_name  \
0              angular/angular   
1  microsoft/microsoft-ui-xaml   
2                dotnet/roslyn   
3              rails/webpacker   

                                               title  \
0                                    Article mistake   
1  Able to change the window height even if IsRes...   
2      Introduce local for 'this' is not very useful   
3  bundle exec rails webpacker:install:typescript...   

                                                body      user_login  \
0  ### Describe the problem that you experienced\...    PiotrZinfWSB   
1  ### Describe the bug\n\nIf you set ExtendsCont...  eduardobragaxz   
2  **Version Used**: VS 15.7\r\n\r\n**Steps to Re...           0xd4d   
3  webpacker version: 4.0.7\r\n\r\nIt appends the...          akoskm   

             created_at                                           answer_1  \
0  2023-05-05T15:54:28Z        Hey @alan-agius4, can I work on this issue?   
1  2024-06-07T20:52:24Z  Hi I'm an AI powered bot that finds similar is...   
2  2018-07-20T15:15:16Z                                 Why is this a bug?   
3  2019-09-09T19:47:02Z  I think this is the missing line:\r\n\r\nhttps...   

                                            answer_2  \
0  @MukulKolpe You can open a PR if you'd like to...   
1  This is very similar to #9670, though that iss...   
2  Design Meeting Notes:\r\n\r\nSince `s` is `rea...   
3  Can this ticket be closed? \r\n\r\nI attempted...   

                                            answer_3  \
0  Following the history and looking at the page,...   
1  This was closed as fixed/duplicate of #7629. T...   
2  @Neme12 is there a good use case for this that...   
3                                                NaN   

                                            answer_4 answer_5      label_1  \
0  This issue has been automatically locked due t...      NaN  help wanted   
1                                                NaN      NaN          bug   
2                                                NaN      NaN          Bug   
3                                                NaN      NaN          bug   

  label_2           label_3        label_4     label_5  likes  
0     NaN  good first issue            NaN         NaN      0  
1     NaN    team-CompInput  area-TitleBar  Regression      0  
2     NaN       help wanted       Area-IDE         NaN      1  
3     NaN       help wanted            NaN         NaN      0

In [9]:
df.tail()

repo_name  \
15950                      glfw/glfw   
15951                 sass/dart-sass   
15952               unisonweb/unison   
15953                flutter/flutter   
15954  bmuschko/gradle-docker-plugin   

                                                   title  \
15950  glfwInit() fails to create a window when on di...   
15951  webpack builds using sass-loader are failing o...   
15952                    Function.fix no longer compiles   
15953              can't read file from android emulator   
15954                     extraHosts should have effect?   

                                                    body    user_login  \
15950  There is no error it just stops the program, b...  Dromader2137   
15951  The fix committed in #1096 is causing the webp...     royson-ct   
15952  The `Function.fix` that exists in base no long...       ceedubs   
15953  I pushed one file into sdcard of emulator, but...        ldbing   
15954  <!--- Provide a brief summary of the issue in ...      Bartman0   

                 created_at  \
15950  2022-07-14T13:52:35Z   
15951  2020-10-06T10:43:49Z   
15952  2023-05-06T18:16:33Z   
15953  2019-09-17T06:52:29Z   
15954  2020-06-12T10:50:35Z   

                                                answer_1  \
15950  That is very strange. If possible, try steppin...   
15951  This is unlikely to be caused by #1096, since ...   
15952  I submitted a [base contribution](https://shar...   
15953  Hello @ldbing !\r\n\r\nThank you for posting a...   
15954  I am not fully sure how Docker Java handles th...   

                                                answer_2  \
15950  Platform is Windows 10, and glfw version is 3....   
15951  Note that we test Sass upstream on CI and ther...   
15952                                                NaN   
15953  Hello @ldbing !\r\n\r\nWithout additional info...   
15954                         What's the status on this?   

                                                answer_3  \
15950  Maybe try disabling the Discord Overlay if it'...   
15951  @nex3 Here is the simple reproduction reposito...   
15952                                                NaN   
15953  This thread has been automatically locked sinc...   
15954  no progress so far.\n\n\n\nRichard Kooijman\nA...   

                                                answer_4  \
15950  Sorry for the really long wait, but the proble...   
15951  You're right, it looks like this was caused by...   
15952                                                NaN   
15953                                                NaN   
15954  OK, I'll close this issue for now. Feel free t...   

                                                answer_5           label_1  \
15950           No worries and thank you for the update!               bug   
15951  Here is a good minimal repro: https://github.c...               bug   
15952                                                NaN               bug   
15953                                                NaN  platform-android   
15954                                                NaN               bug   

      label_2                        label_3   label_4 label_5  likes  
15950     NaN                        Windows  external     NaN      0  
15951     NaN                     needs info       NaN     NaN      4  
15952     NaN                    typechecker       NaN     NaN      0  
15953     NaN  waiting for customer response       NaN     NaN      0  
15954     NaN                            NaN       NaN     NaN      0

#### Max Likes Which A Repository Contain

In [10]:
df['likes'].max()

46

In [11]:
df['likes'].min()

0

##### Top 5 likes which repositories contain

In [12]:
top_3 = df['likes'].nlargest(5)

print(f'These are top five values of like which repositories contain')
i = 1
for likes in top_3:
    print(f'{i}. {likes}')
    i += 1
del i

These are top five values of like which repositories contain
1. 46
2. 25
3. 24
4. 22
5. 21


**Note:** `As most of the repositories contain 0 likes so we don't need to use nsmallest()`

#### We can change the datatype of date, As it is shown in object

In [13]:
df['created_at'] = pd.to_datetime(df['created_at'])

#### Now check the info

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15955 entries, 0 to 15954
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype              
---  ------      --------------  -----              
 0   repo_name   15955 non-null  object             
 1   title       15955 non-null  object             
 2   body        15955 non-null  object             
 3   user_login  15955 non-null  object             
 4   created_at  15955 non-null  datetime64[ns, UTC]
 5   answer_1    15955 non-null  object             
 6   answer_2    12508 non-null  object             
 7   answer_3    9006 non-null   object             
 8   answer_4    6435 non-null   object             
 9   answer_5    4513 non-null   object             
 10  label_1     15946 non-null  object             
 11  label_2     4 non-null      object             
 12  label_3     8438 non-null   object             
 13  label_4     3783 non-null   object             
 14  label_5     1546 non-null   object    

#### Show first 5 values of data

In [15]:
df.head()

repo_name  \
0              angular/angular   
1  microsoft/microsoft-ui-xaml   
2                dotnet/roslyn   
3              rails/webpacker   
4             google/ExoPlayer   

                                               title  \
0                                    Article mistake   
1  Able to change the window height even if IsRes...   
2      Introduce local for 'this' is not very useful   
3  bundle exec rails webpacker:install:typescript...   
4  MediaSession play/pause events not propagated ...   

                                                body      user_login  \
0  ### Describe the problem that you experienced\...    PiotrZinfWSB   
1  ### Describe the bug\n\nIf you set ExtendsCont...  eduardobragaxz   
2  **Version Used**: VS 15.7\r\n\r\n**Steps to Re...           0xd4d   
3  webpacker version: 4.0.7\r\n\r\nIt appends the...          akoskm   
4  ### ExoPlayer Version\r\n\r\n2.18.0\r\n\r\n###...      TheBeastLT   

                 created_at  \
0 2023-05-05 15:54:28+00:00   
1 2024-06-07 20:52:24+00:00   
2 2018-07-20 15:15:16+00:00   
3 2019-09-09 19:47:02+00:00   
4 2022-07-08 05:11:40+00:00   

                                            answer_1  \
0        Hey @alan-agius4, can I work on this issue?   
1  Hi I'm an AI powered bot that finds similar is...   
2                                 Why is this a bug?   
3  I think this is the missing line:\r\n\r\nhttps...   
4  Thank you for reporting this. We submitted a f...   

                                            answer_2  \
0  @MukulKolpe You can open a PR if you'd like to...   
1  This is very similar to #9670, though that iss...   
2  Design Meeting Notes:\r\n\r\nSince `s` is `rea...   
3  Can this ticket be closed? \r\n\r\nI attempted...   
4                                                NaN   

                                            answer_3  \
0  Following the history and looking at the page,...   
1  This was closed as fixed/duplicate of #7629. T...   
2  @Neme12 is there a good use case for this that...   
3                                                NaN   
4                                                NaN   

                                            answer_4 answer_5      label_1  \
0  This issue has been automatically locked due t...      NaN  help wanted   
1                                                NaN      NaN          bug   
2                                                NaN      NaN          Bug   
3                                                NaN      NaN          bug   
4                                                NaN      NaN          bug   

  label_2           label_3        label_4     label_5  likes  
0     NaN  good first issue            NaN         NaN      0  
1     NaN    team-CompInput  area-TitleBar  Regression      0  
2     NaN       help wanted       Area-IDE         NaN      1  
3     NaN       help wanted            NaN         NaN      0  
4     NaN               NaN            NaN         NaN      0